php/src/Communicator.cpp

```diff
@@ -209,28 +209,25 @@ ZEND_METHOD(Ice_Communicator, destroy)
     assert(c);
     CommunicatorMap* m = reinterpret_cast<CommunicatorMap*>(ICE_G(communicatorMap));
     assert(m);
-    if (m->find(c) != m->end())
```

Comment:
This lookup doesn't make sense, the communicator must be in the communicatorMap for the request if destroy is being called.

Reply:
what if you call destroy twice on the same communicator in the same PHP request?

Reply:
I assume that with your test, `m->erase(c)` returns 0 for the second destroy.